pkg/flow/flow.go

```diff
@@ -209,9 +214,66 @@ func (c *Flow) LoadFile(f *File) error {
 	return diags.ErrorOrNil()
 }
 
+// ComponentInfos returns the component infos.
+func (c *Flow) ComponentInfos() []*river.ComponentField {
+	cns := c.loader.Components()
+	infos := make([]*river.ComponentField, len(cns))
+	refByBacktrack := make(map[string]*river.ComponentField, 0)
+	for i, com := range cns {
+		refs, _ := controller.ComponentReferences(com, c.loader.Graph())
```

Comment:
Can you add a TODO here to change this to use a graph that doesn't have transitive reductions applied? That'll produce some unexpected results for users.

Reply:
Cached the edges pre-reduce for later usage.